vllm/worker/habana_model_runner.py

```diff
@@ -1008,6 +1034,19 @@ def warmup_model(self, kv_caches: List[torch.Tensor]) -> None:
         logger.info(f"Warmup finished in {elapsed_time:.0f} secs, allocated {format_bytes(end_mem - start_mem)} of device memory")
         self.profiler.end()
 
+    def shutdown_hqt(self):
+        print('hqt shutdown')
+        if model_config := getattr(self, "model_config", None):
+            if getattr(model_config, "quantization", None) == 'hqt':
+                print('hqt shutdown start')
+                import habana_quantization_toolkit
+                if habana_quantization_toolkit is not None:
+                    habana_quantization_toolkit.finish_measurements(self.model.model)
```

Comment:
self.model.model might not exist here

Reply:
Why?
you wrap it with HpuModelAdapter anyway.